url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/cmdlib/FileLibrary.java

```diff
@@ -0,0 +1,301 @@
+{
+    private static final Pattern DRIVE = Pattern.compile("^\\p{Alpha}:.*$");
+    private File currentDirectory;
+
+    // Always quit. The option is allowed but does not have any effect.
+    //
+    @ScriptyCommand(name = "cd")
+    @ScriptyStdArgList(
+            optional = {
+                    @ScriptyArg(name = "file", type = "OneOf (String) (Instance java.io.File)", value = ".")},
+            named = {
+                    @ScriptyArg(name = "quiet", type = "Boolean", value = "false", optional = true)
+            })
+    public File cd(
+            @ScriptyParam("file") Object aFileRepr)
+            throws CommandException
+    {
+        File lFile;
+        if (aFileRepr instanceof File) lFile = (File) aFileRepr;
+        else lFile = resolveFile((String) aFileRepr);
+        if (!lFile.isDirectory())
+            throw new CommandException(String.format("Cannot go to '%s', it is not a directory.", lFile));
+        else currentDirectory = lFile;
+        return lFile;
+    }
+
+    // Print the currrent directory and return the File object.
+    //
+    @ScriptyCommand(name = "pwd")
+    @ScriptyRefArgList(ref = "noarg + quiet")
+    public File pwd(@ScriptyParam("quiet") boolean aQuiet, @ScriptyBindingParam("*output") PrintWriter aWriter)
+            throws CommandException
+    {
+        final File lCurDir = getCurrentDirectory();
+
+        try
+        {
+            if (!aQuiet) aWriter.println(lCurDir.getCanonicalPath());
+            return lCurDir;
+        } catch (IOException e)
+        {
+            throw new CommandException(String.format("Cannot calculate the canonical name for '%s'.", lCurDir), e);
+        }
+
+    }
+
+    // Print a listing of the current directory and return it als an array.
+    //  - grep=regexp: applied to the absolute pathname.
+    //  - quiet=true|*false.
+    //  - files=*true|false : include files.
+    //  - dirs=*true|false : include directories.
+    //  - exec=lambda: process the files using a lambda. Processing is done after grepping and filtering.
+    //
+    @ScriptyCommand(name = "ls")
+    @ScriptyStdArgList(
+            optional = {
+                    @ScriptyArg(name = "file", type = "OneOf (String) (Instance java.io.File)", value = ".")},
+            named = {
+                    @ScriptyArg(name = "quiet", type = "Boolean", value = "false", optional = true),
+                    @ScriptyArg(name = "files", type = "Boolean", value = "true", optional = true),
+                    @ScriptyArg(name = "dirs", type = "Boolean", value = "true", optional = true),
+                    @ScriptyArg(name = "grep", type = "String", value = ".*", optional = true),
+                    @ScriptyArg(name = "exec", type = "Instance com.sdicons.scripty.parser.Lambda nullAllowed=true", value = "{null}", optional = true),
+                    @ScriptyArg(name = "recursive", type = "Boolean", value = "false", optional = true)
+            })
+    public Object[] ls(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.